xargs/main.cpp

```diff
@@ -0,0 +1,41 @@
+#include <unistd.h>
+#include <bits/stdc++.h>
+#include <iostream>
+#include <string>
+#include <vector>
+
+
+int main(int argc, char **argv)
+{
+    std::vector<std::string> arguments(100);
+    int i = 0;
+    std::cin >> arguments[i];
+    while(arguments[i] != "CTRL+D")
+    {
+        ++i;
+        std::cin >> arguments[i];
+    }
+
+    char** args = new char*[arguments.size()];
+    for (int j = 0; j < arguments.size(); j++)
+    {
+        args[j] = const_cast<char*>(arguments[j].c_str());
```

Comment:
No. Never. In no circumstances use `const_cast`. You don't even need to. Do this instead:
args[j] = strdup(arguments[j].c_str()); // You need to `free(args[j]);` later.

Reply:
Done.